aten/src/ATen/EmptyTensor.cpp

```diff
@@ -98,6 +98,27 @@ size_t computeStorageNbytes(
 #endif
 }
 
+// not including mobile-only macros in this function,
+// since mobile shouldn't be using symints.
+SymInt computeStorageNbytesSymInt(
```

Comment:
it would be nice to get `aten::empty` using this helper too

Reply:
Yep... it looks like it might require some larger changes to `PySymInt` et.al, so I figured I wouldn't bite it off in this PR. LMK what you think though.


It looks like the way that `SymInt::operator+` (and the others) are implemented today, if one SymInt is symbolic and the other isn't, it'll effectively promote the non-symint to a real `SymbolicIntNode` object. Which I don't think is what we want here? If `at::empty()` is meant to construct a contiguous tensor, it seems weird to introduce a bunch of extra sym stride nodes in the expression to compute the number of bytes.

Reply:
Why isn't it what you want? If a size is symbolic you are going to need a symbolic storage size too.

Reply:
What I mean is, the symbolic expression computed by `at::empty` today looks something like:
```
storage_size = size * size2 * size3 * ...
```

And the symbolic expression you'll get from running `computeStorageNbytesSymInt` looks a lot more complicated, since it involves symbolic strides:
```
(storage_offset) + 1 + ([size1 - 1] * stride1) + ([size2 - 1] * stride2) + ([size3 - 1] * stride3) + ...
```

Is the idea that this is ok, because it should be possible to simplify that expression back down to the contiguous version? Since `strideN` variables above would actually be written in terms of sizes.

Reply:
Oh, this is fair. I suppose a simplified path when the strides are contiguous is a reasonable idea.

Reply:
In principle if you guard the strides enough (as they are contiguous), yes they should be equivalent, but we'd have to check if sympy can actualy figure it out.